Multigeo site enumeration API support (#19923)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/site-getallsites.md

---
title: "sites: getAllSites"
description: "List all root sites across geographies in an organization - OneDrive API"
author: tushar20
ms.author: tkanaujia
ms.date: 5/3/2023
ms.localizationpriority: medium
ms.prod: "sharepoint"
doc_type: apiPageType
---

# sites: getAllSites

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

List all available [sites](../resources/site.md) across geographies in an organization.

> For more details, see [Best practices for discovering files and detecting changes at scale](../concepts/scan-guidance?view=odsp-graph-online.md).

Reviewer: ```suggestion
For more details, see [Best practices for discovering files and detecting changes at scale](../onedrive/developer/rest-api/concepts/scan-guidance.md).
```

> For more details, see [Best practices for discovering files and detecting changes at scale](/onedrive/developer/rest-api/concepts/scan-guidance.md).

Reviewer: ```suggestion
For more details, see [Best practices for discovering files and detecting changes at scale](/onedrive/developer/rest-api/concepts/scan-guidance?view=odsp-graph-online.md).
```

> For more details, see [Best practices for discovering files and detecting changes at scale](../onedrive/developer/rest-api/concepts/scan-guidance.md).

Reviewer: ```suggestion
For more details, see [Best practices for discovering files and detecting changes at scale](https://learn.microsoft.com/en-us/onedrive/developer/rest-api/concepts/scan-guidance).
```

> For more details, see [Best practices for discovering files and detecting changes at scale](/onedrive/developer/rest-api/concepts/scan-guidance?view=odsp-graph-online.md).

Reviewer: ```suggestion
For more details, see [Best practices for discovering files and detecting changes at scale](../concepts/scan-guidance?view=odsp-graph-online.md).
```

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

### List all site collections

|Permission type                        | Permissions (from least to most privileged)
|:--------------------------------------|:-------------------------------------
|Delegated (work or school account)     | Not supported.
|Delegated (personal Microsoft account) | Not supported.
|Application                            | Sites.Read.All, Sites.ReadWrite.All

## HTTP request

<!-- { "blockType": "ignored" } -->

```http

> GET /sites/getAllSites

Reviewer: ```suggestion
GET /site/getAllSites
```

```

## Example

### Request

# [HTTP](#tab/http)

<!-- { "blockType": "ignored" } -->

```msgraph-interactive

> GET /sites/getAllSites

Reviewer: ```suggestion
GET /site/getAllSites
```
